lib/vmpooler/pool_manager.rb

```diff
@@ -746,7 +764,12 @@ def execute!
           end
         end
 
-        sleep(1)
+        sleep(loop_delay)
+
+        unless maxloop.zero?
+          break if loop_count >= maxloop
+          loop_count = loop_count + 1
+        end
       end
     end
   end
```

Comment:
Yes.  Perhaps my commit title is misleading.

But in the commit msg itself:
```
Previously the check_pool method would execute the loop indefinitely as it did not
have a terminating condition.  This made it impossible to test.  This commit
modifies the check_pool method so that it can take a maxloop and delay parameter
so that it can be tested.
```